test/framework/http.go

```diff
@@ -88,13 +103,77 @@ func ExecuteHTTPRequestWithUnmarshalledResponse(namespace, httpMethod, uri, path
 
 // IsHTTPRequestSuccessful makes and checks whether an http request is successful
 func IsHTTPRequestSuccessful(namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
-	response, err := ExecuteHTTPRequest(namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+	return IsHTTPRequestSuccessfulC(&http.Client{}, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+}
+
+// IsHTTPRequestSuccessful makes and checks whether an http request is successful using a given client
+func IsHTTPRequestSuccessfulC(client *http.Client, namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
+	response, err := ExecuteHTTPRequestC(client, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
 	if err != nil {
 		return false, err
 	}
+	io.Copy(ioutil.Discard, response.Body)  // Just read the response to be able to close the connection properly
+	defer response.Body.Close()
 	return checkHTTPResponseSuccessful(namespace, response), nil
 }
 
+// ExecuteHTTPRequestsInThreads executes given number of requests using given number of threads (Go routines).
+// Returns []HTTPRequestResult with the outcome of each thread (HTTPRequestResultSuccess or HTTPRequestResultError).
+// Returns error if the desired number of requests cannot be precisely divided to the threads.
+// Useful for performance testing.
+func ExecuteHTTPRequestsInThreads(namespace, httpMethod string, requestCount, threadCount int, routeURI, path, bodyFormat, bodyContent string) ([]HTTPRequestResult, error) {
+	if requestCount % threadCount != 0 {
+		return nil, fmt.Errorf("Cannot precisely divide %d requests to %d threads. Use different numbers.", requestCount, threadCount)
+	}
+	requestPerThread := requestCount / threadCount
+	results := make([]HTTPRequestResult, threadCount)
+	waitGroup := &sync.WaitGroup{}
+	waitGroup.Add(threadCount)
+
+	GetLogger(namespace).Info("Starting request threads")
+	startTime := time.Now()
+
+	for threadId := 0; threadId < threadCount; threadId++ {
+		client := createCustomClient()
+		go runRequestRoutine(threadId, waitGroup, client, namespace, httpMethod, requestPerThread, routeURI, path, bodyFormat, bodyContent, results)
```

Comment:
The same for all other functions that you're receiving all those parameters to execute a request. :stuck_out_tongue_closed_eyes:

Reply:
Well, that would need a complete overhaul of the framework because if I did it just for this method, then it would be inconsistent with the rest of the methods.
The only thing I am adding in my PR is the number of requests and number of threads. Shouldn't we address this outside of this PR?

Reply:
+1 for me to address this in another PR

Reply:
+1 also

Reply:
np, please open a follow up JIRA then to arrange all those methods parameters.

Reply:
https://issues.redhat.com/browse/KOGITO-1729 opened, merging this one.